File: server-laravel-hybrid/app/Policies/PersonalProjectPolicy.php
<?php

namespace App\Policies;

use Rhino\Policies\ResourcePolicy;
use Illuminate\Contracts\Auth\Authenticatable;

/**
 * PersonalProjectPolicy — personal group is user-owned (no roles).
 *
 * With group-membership enforcement ON, the personal user resolves permissions
 * from their `route_group = 'personal'` (org-less) user_roles row. The policy's
 * checkPermission delegates to User::hasPermission with that route_group, so the
 * membership row's `permissions` blob gates CRUD. Ownership isolation is handled
 * by PersonalProjectScope.
 */
class PersonalProjectPolicy extends ResourcePolicy
{
    public function permittedAttributesForShow(?Authenticatable $user): array
    {
        return $user ? ['*'] : [];
    }

    public function permittedAttributesForCreate(?Authenticatable $user): array
    {
        return $user ? ['title', 'description', 'status'] : [];
    }

    public function permittedAttributesForUpdate(?Authenticatable $user): array
    {
        return $this->permittedAttributesForCreate($user);
    }
}

File: server-rails-hybrid/app/policies/personal_project_policy.rb
# frozen_string_literal: true

# PersonalProjectPolicy — the `personal` group's user-owned model.
#
# Action-level authorization runs through the normal permission machinery
# (the personal membership row grants `personal-projects.*`). Row isolation is
# enforced by Scopes::PersonalProjectScope, so the policy is permissive on
# attributes.
class PersonalProjectPolicy < Rhino::ResourcePolicy
  def permitted_attributes_for_show(_user) = ["*"]
  def hidden_attributes_for_show(_user) = []
  def permitted_attributes_for_create(_user) = %w[title description status]
  def permitted_attributes_for_update(_user) = %w[title description status]
end
